build_tools/docker/Dockerfile

```diff
@@ -0,0 +1,40 @@
+ARG BASE_IMG=ubuntu:18.04
+FROM ${BASE_IMG} as dev-base
+
+ARG ARCH="x86_64"
+ARG TARGETARCH="amd64"
+ARG BAZEL_VERSION=4.2.1
+
+# Install basic packages
+RUN apt-get update                                              && \
+    apt-get install -y                                             \
+    python3.8                                                      \
+    python3.8-dev                                                  \
+    cmake                                                          \
+    ccache                                                         \
+    ninja-build                                                    \
+    git                                                            \
+    python3-pip                                                    \
+    wget                                                           \
+    clang-10                                                       \
```

Comment:
@asaadaldien PTAL. Previously we'd decided to not send this upstream because clang on GHA VMs was already new, but with Ubuntu 18 container the default clang is old and fails with the AVX errors.

Reply:
Actually, this breaks CMake workflow because it is looking for `clang` and not `clang-10` :) Looking into a way to symbolic link so as to not update every `-DCMAKE_C_COMPILER=clang` with `-DCMAKE_C_COMPILER=clang-10`.

Reply:
Looks like this is what I need: https://man7.org/linux/man-pages/man1/update-alternatives.1.html (and we're already doing this for python).